src/test/java/spoon/test/template/SubstitutionTest.java

```diff
@@ -0,0 +1,60 @@
+package spoon.test.template;
+
+import org.junit.jupiter.api.Test;
+import spoon.Launcher;
+import spoon.reflect.code.CtExpression;
+import spoon.reflect.declaration.CtField;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.factory.Factory;
+import spoon.support.compiler.FileSystemFile;
+import spoon.template.StatementTemplate;
+import spoon.template.Substitution;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SubstitutionTest {
+
+    public class SingleFieldTemplate extends StatementTemplate {
```

Comment:
Sorry, one thing I missed. This is test input that is only relevant to a single test here, so it would be prudent to make it `private` and `static` (private makes it inaccessible to other test classes, static detaches it from the state of the enclosing class).

```suggestion
    private static class SingleFieldTemplate extends StatementTemplate {
```